File: app/src/main/java/com/hlibrary/util/command/CommandResult.java
package com.hlibrary.util.command;

/**
 * Created by linwenhui on 2018/3/5.
 */

public class CommandResult {
    /** result of command **/
    public int exitCode;
    /** success message of command result **/
    public String successMsg;
    /** error message of command result **/
    public String errorMsg;

    public CommandResult(int result) {
        this.exitCode = result;
    }

    public CommandResult(int result, String successMsg, String errorMsg) {
        this.exitCode = result;
        this.successMsg = successMsg;
        this.errorMsg = errorMsg;
    }

    @Override
    public String toString() {
        return "exitCode=" + exitCode + "; successMsg=" + successMsg
                + "; errorMsg=" + errorMsg;
    }
}

File: app/src/main/java/com/hlibrary/util/command/CommandResult.kt
package com.hlibrary.util.command

/**
 *
 * @author linwh
 * @date 2018/3/5
 */
class CommandResult {
    /** result of command  */
    var exitCode: Int = -1
    /** success message of command result  */
    var successMsg: String = ""
    /** error message of command result  */
    var errorMsg: String = ""

    constructor(result: Int) {
        this.exitCode = result
    }

    constructor(result: Int, successMsg: String, errorMsg: String) {
        this.exitCode = result
        this.successMsg = successMsg
        this.errorMsg = errorMsg
    }

    override fun toString(): String {
        return ("exitCode=" + exitCode + "; successMsg=" + successMsg
                + "; errorMsg=" + errorMsg)
    }
}
